### ISSUE TITLE
Idea: Optional pre-built translation files for popular languages

### ISSUE THREAD

**Issue by mor10** [Automattic/_s#726]
This is just an idea: Many _s users probably don't make a lot of changes to the translated strings in the theme. I'm thinking it would be useful if there was some sort of repo where _s users could pull pre-built translation files for popular languages that could then be baked into their own custom themes. This should be relatively easy to build out and maintain and would encourage more theme developers to ship their themes with language support.

**Comment by emiluzelac** [comment]
To offer under "Advanced Options" maybe?

**Comment by leolopes** [comment]
I find it a very good idea. I had a Brazilian Portuguese translation for an
older version and offered it to help, but one of the authors didn't want to
add it to the repo at the moment.
I understand that, but with an exclusive repo that would be a no-brainer.

**Comment by jamigibbs** [comment]
Doesn't the .pot file generate string locations based on the line number of the template file? So if even a single line of the template file is added/removed, the translation file would need to be generated again by the user?

**Comment by leolopes** [comment]
​That would be true, but it doesn't nullify the benefit of having a head
start.​

**Comment by emiluzelac** [comment]
If `_s` would do something like this https://github.com/devinsays/underscores-grunt/blob/master/Gruntfile.js the generate issue would be eliminated, not to mention huge time saver as well.

**Comment by mor10** [comment]
@jamigibbs When you change the structure or add new strings you have to regenerate the .pot file and whatever app you use to write out the actual translations will do its best to figure out what happened and remap all the translation files. I use Poedit for this and make a lot of changes that permeate through 17 different translation files without issues. The point of this proposal would be to provide developers with an already built-out starting point for translation files so each individual dev doesn't have to go out and source translations themselves.

**Comment by ptbello** [comment]
@mor10 the developers decided not to include translations (see #50), the argument being that _s is meant to be forked and changed, and so would the translatable strings; however I agree 100% it would be helpful to have starter translations that can be edited.  
So following your suggestion, I created a repo just for that https://github.com/ptbello/_slated  
and added the Italian translation which I use and mantain.

Other languages I've found so far:

- @leolopes mentioned having Brazilian Portoguese; also mentioned here #718 by @pedro-mendonca
- @fxbenard mentioned having a French translation here: #50
- @lauretum mentioned Dutch here: #151
- @MHM5000 mentioned Farsi here: #464
- @naokomc mentioned Japanese here: #111 and @Satojii here: #422

If you're getting mentioned, please take a few minutes to refresh your translation to the current .pot and submit a pr to https://github.com/ptbello/_slated.

**Comment by mor10** [comment]
Yes! This is great. I will submit a Norwegian translation presently.

In medias res on my paranoid android

**Comment by pedro-mendonca** [comment]
Great idea, I'll pull my files of pt_PT.

**Comment by pedro-mendonca** [comment]
Is the .pot file synced with original [_s.pot](https://github.com/Automattic/_s/blob/master/languages/_s.pot) file?

**Comment by ptbello** [comment]
@pedro-mendonca yes, as of yesterday. And thanks for the pr

**Comment by Zodiac1978** [comment]
There is a German translation from Caspar (@glueckpress) and Bernhard (@2ndkauboy):
https://github.com/glueckpress/_s/tree/_s-de_DE/languages

**Comment by corvannoorloos** [comment]
What about adding _s to https://translate.wordpress.org/projects?

**Comment by karmatosed** [comment]
I don't think this is a good idea. _s isn't a framework. It's a starter theme. People are meant to add their own content and create their own theme. This right off makes any .pot included invalid and need to be regenerated.

My main point against on top of this is maintenance. It's just something that would spiral.

In saying this, if someone wanted to create a repo and have their own _s translations in it and do this for community - awesome. I do not see it as something in the scope of the _s project though.

**Comment by glueckpress** [comment]
What @karmatosed says. Though arguably many strings never will be touched for customization during the average theme project, languages files in this repository would immediately raise the expectation that they’ll be well maintained. I don’t think that expectation would be met in the long run. 

Regarding @Zodiac1978’s example: My (outdated) German translation was found via Google search, forked, updated and a pull request was sent. Imo that’s how it should go with _s and translations.

**Comment by ptbello** [comment]
@karmatosed, @glueckpress I agree.
My personal experience is that when I needed an Italian translation, I desperately scoured the interwebs for quite a while before resolving to painfully translate every string from scratch.
On that occasion, if I had found even an outdated version as a starting point it would have saved me some time, and I would have been more than happy to updated it myself and post a pr back to where I'd found it.
To sum up - outdated is better than none at all, as long as expectations aren't raised, so I suggest
when people offer (or ask) for translations here, you could point them to this thread for context and to https://github.com/ptbello/_slated for contributing.
Also, you could open issues  there [@]mentioning anyone who has missing translations you happen to already know of.

**Comment by 2ndkauboy** [comment]
@ptbello I was in the exact same situation as you. Luckily I found the (outdated) translation of @glueckpress before I was half way through. So I forked his repo, added the new strings and opened a PR.

I can also understand why the `_s ` doesn't want to have translations included. But I agree with @glueckpress that as the strings rarely change, it would be great to have translation files shipped with a theme, even if it is a bit outdated.

I would even go a step further and would like to have a translation base with common string found in themes. Every theme will have a string e.g. "Oops! That page can't be found.", so why not have a translation database to get translation for such string. It would than be quite easy to build an up tp date translation file for `_s` or any other theme.

For `_s` specifically, I would love to see a central repo, that has translation files ready to download. We might even use a GlotPress or something similar for this, so everyone could easily contribute, even if he/she doesn't know how to use Github or has only time to translate some strings.

For the underscores.me I would than love to see a translation option in the advanched options view, so one can easily download a package including the proper (hopefully up to date) translation file.

**Comment by ptbello** [comment]
@2ndkauboy nice suggestions.
I think the first step could be that people searching for an _s translation at least find out there's a repo for that, and contribute if their translation is missing or outdated.
Looking forward to your pr for german.

**Comment by 2ndkauboy** [comment]
@ptbello: I'll try to create the pull request later, when I return home.

**Comment by MHM5000** [comment]
You'll have Farsi in _slated by tomorrow...

**Comment by ptbello** [comment]
@2ndkauboy merged, über cool!  
@MHM5000 thanks, looking forward to having farsi.

**Comment by fklein-lu** [comment]
Closing this, as this _s related issue is resolved.
